File: rustCode/problems/src/bin/lc121.rs
// https://leetcode.com/problems/best-time-to-buy-and-sell-stock/

fn max_profit(prices: Vec<i32>) -> i32 {

    let mut ans = 0;
    let mut m = prices[0];

    for i in 1..prices.len() {
        ans = std::cmp::max(ans, prices[i] - m);
        m = std::cmp::min(m, prices[i]);
    }

        
    ans
}

fn main() {
    
}

File: python/problems/lc121.py
from collections import deque
from typing import List, Deque







class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        ans = 0
        curMin = prices[0]
        for i in prices:
            if i < curMin:
                curMin = i
            ans = max(ans, i - curMin)


        return ans






def main():
    # prices = [7,1,5,3,6,4]
    prices = [7,6,4,3,1]
    solution = Solution()

    print(f"ans is {solution.maxProfit(prices)}")

if __name__ == '__main__':
    main()
